MorseDecodeEncode/src/app/decode/decode.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component } from '@angular/core';
+import { DecodeService } from './decode.service';
+
+@Component({
+  selector: 'app-decode',
+  templateUrl: './decode.component.html',
+  styleUrls: ['./decode.component.css']
+})
+//morse -> text
+export class DecodeComponent {
+  public input: string;
+  public result: string;
+
+  constructor(private decodeService: DecodeService) {
+    this.input = '';
+    this.result = '';
+  }
+
+  public onDecodeBtn(){
+    this.result = this.decodeService.decode(this.input);
+  }
+
+  public isValidInput(): boolean {
+    if (this.input.trim().length > 0) {
```

Comment:
Here you could use a regex if you want (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions)